### PR TITLE
Libcurl prep v2

### PR DESCRIPTION
**Pull request by cgwalters**

### CODE REVIEW

File: src/libostree/ostree-fetcher.c [hunk 1]
```diff
@@ -63,7 +63,6 @@ typedef struct {
   int max_outstanding;
 
   /* Queue for libsoup, see bgo#708591 */
```
Comment (jlebon) [inline src/libostree/ostree-fetcher.c]: Outdated comment?

File: src/libostree/ostree-fetcher.c [hunk 2]
```diff
@@ -77,8 +76,10 @@ typedef struct {
 
 } ThreadClosure;
 
+
 static void
-session_thread_process_pending_queue (ThreadClosure *thread_closure);
+start_pending_request (ThreadClosure *thread_closure,
+                       GTask         *task);
```
Comment (jlebon) [inline src/libostree/ostree-fetcher.c]: Do we actually need this prototype anymore?

File: src/libostree/ostree-fetcher.c [hunk 3]
```diff
@@ -1110,10 +1081,8 @@ on_request_sent (GObject        *object,
                 goto out;
 
               (void) g_input_stream_close (pending->request_body, NULL, NULL);
-              g_queue_insert_sorted (&pending->thread_closure->pending_queue,
-                                     g_object_ref (task), pending_task_compare,
-                                     NULL);
-              remove_pending_rerun_queue (pending);
```
Comment (jlebon) [inline src/libostree/ostree-fetcher.c]: We can probably delete this function now and remove the pending object right at the call sites. (Or at least, rename it to just `remove_pending`).

**Comment by cgwalters** [comment]
Ok, comments addressed ⬆️

**Comment by jlebon** [comment]
@rh-atomic-bot r+ 7d05650

**Pull request closed without merging**